test/core/parallelAlgorithms.cpp

```diff
@@ -0,0 +1,102 @@
+// SPDX-License-Identifier: MIT
+// SPDX-FileCopyrightText: 2023 NeoFOAM authors
+
+#define CATCH_CONFIG_RUNNER // Define this before including catch.hpp to create
+                            // a custom main
+#include <catch2/catch_session.hpp>
+#include <catch2/catch_test_macros.hpp>
+#include <catch2/generators/catch_generators_all.hpp>
+
+#include "NeoFOAM/fields/field.hpp"
+#include "NeoFOAM/core/executor/executor.hpp"
+#include "NeoFOAM/core/parallelAlgorithms.hpp"
+
+
+TEST_CASE("parallelFor")
+{
+    NeoFOAM::Executor exec = GENERATE(
+        NeoFOAM::Executor(NeoFOAM::CPUExecutor {}),
+        NeoFOAM::Executor(NeoFOAM::OMPExecutor {}),
+        NeoFOAM::Executor(NeoFOAM::GPUExecutor {})
+    );
+    std::string execName = std::visit([](auto e) { return e.print(); }, exec);
+
+    SECTION("parallelFor_" + execName)
+    {
+        NeoFOAM::Field<NeoFOAM::scalar> fieldA(exec, 5);
+        NeoFOAM::fill(fieldA, 0.0);
+        NeoFOAM::Field<NeoFOAM::scalar> fieldB(exec, 5);
+        auto spanA = fieldA.span();
+        auto spanB = fieldB.span();
+        NeoFOAM::fill(fieldB, 1.0);
+        NeoFOAM::parallelFor(
+            exec, {0, 5}, KOKKOS_LAMBDA(const size_t i) { spanA[i] = spanB[i] + 2.0; }
+        );
+        auto hostSpanA = fieldA.copyToHost().span();
+        for (int i = 0; i < 5; i++)
+        {
+            REQUIRE(hostSpanA[i] == 3.0);
+        }
+    }
+
+    SECTION("parallelFor_Field_" + execName)
+    {
+        NeoFOAM::Field<NeoFOAM::scalar> fieldA(exec, 5);
+        NeoFOAM::fill(fieldA, 0.0);
+        NeoFOAM::Field<NeoFOAM::scalar> fieldB(exec, 5);
+        auto spanA = fieldA.span();
+        auto spanB = fieldB.span();
+        NeoFOAM::fill(fieldB, 1.0);
+        NeoFOAM::parallelFor(
+            fieldA, KOKKOS_LAMBDA(const size_t i) { return spanB[i] + 2.0; }
+        );
+        auto hostSpanA = fieldA.copyToHost().span();
+        for (int i = 0; i < 5; i++)
+        {
+            REQUIRE(hostSpanA[i] == 3.0);
+        }
```

Comment:
Same here, rather use a range based loop.

```suggestion
        for (int i = 0; i < 5; i++)
        {
            REQUIRE(hostSpanA[i] == 3.0);
        }
```

Reply:
done